main.js

```diff
@@ -0,0 +1,334 @@
+    };
+
+    let tempAcc = 0;
+    let icons = {};
+    let weatherDescriptions = {};
+
+    // identify the appropriate data range
+    let indexRange = [];
+
+    if (day === 1) {
+      indexRange = [0, 7];
+    }
+    if (day === 2) {
+      indexRange = [8, 15];
+    }
+    if (day === 3) {
+      indexRange = [16, 23];
+    }
+    if (day === 4) {
+      indexRange = [24, 31];
+    }
+    if (day === 5) {
+      indexRange = [32, 39];
+    }
+
+    // loop through the identified data range to accumulte data
+    for (let i = indexRange[0]; i <= indexRange[1]; i++) {
+      const time = data.list[i];
+
+      // accumulate temperature values to calculate average after the fact
+      tempAcc += time.main.temp;
+
+      // accumulate weather descriptions
+      if (weatherDescriptions.hasOwnProperty(time.weather[0].main)) {
+        weatherDescriptions[time.weather[0].main]++;
+      } else {
+        weatherDescriptions[time.weather[0].main] = 1;
+      }
+
+      // accumulate weather icons
+      if (icons.hasOwnProperty(time.weather[0].icon)) {
+        icons[time.weather[0].icon]++;
+      } else {
+        icons[time.weather[0].icon] = 1;
+      }
+    }
+
+    // calculate average by dividing accumulated temperatures by 8
+    dayData.avgTemp = (tempAcc / 8).toFixed(1);
+
+    // find the day of the week by creating a new Date, and the processessing that with Intl.DateTimeFormat.
+
+    const dt_txt = new Date(data.list[indexRange[0]].dt_txt);
+
+    const dateFormatter = new Intl.DateTimeFormat("en-US", {
+      weekday: "long",
+    });
+
+    dayData.dayName = dateFormatter.format(dt_txt);
+
+    // find most common weather description. if all weather descriptions are equal, leave the first one.
+    let weatherDescriptionsMaxCount = 0;
+
+    for (let key in weatherDescriptions) {
+      if (weatherDescriptions[key] > weatherDescriptionsMaxCount) {
+        weatherDescriptionsMaxCount = weatherDescriptions[key];
+        dayData.weatherSummary = key;
+      }
+    }
+
+    // find most common icon. if all icons are equal, leave the first icon.
+    let iconMaxCount = 0;
+
+    for (let key in icons) {
+      if (icons[key] > iconMaxCount) {
+        iconMaxCount = icons[key];
+        dayData.iconSummary = key;
+      }
+    }
+
+    return dayData;
+  }
+
+  //loop through all days to return a summary data object or each day
+  for (let i = 1; i <= 5; i++) {
+    const processedDayData = processDayData(i);
+    dividedDayData.push(processedDayData);
+  }
+
+  return dividedDayData;
+}
+
+async function addFiveDay(summaryData) {
+  const fiveDaySection = document.querySelector(".forecast-5-day");
+  fiveDaySection.replaceChildren();
+
+  const template = `
+    <div class="row justify-content-center">
+      <h2>Five Day Forecast</h2>
+    </div>
+
+
+    <div class="row justify-content-center my-2">
+      <div class="col-md-2 container-fluid border justify-content-center" id="day-1">
+        <p class="text-center mt-2">
```

Comment:
You could make a function that is a template of just the day of summary data and then iterate over it. Would be a lot less code and you would repeat it less

Reply:
Did this!